functional-tests/fn_bridge_withdraw_happy.py

```diff
@@ -0,0 +1,179 @@
+
+        rollup_deposit_amount = DEFAULT_ROLLUP_PARAMS["deposit_amount"]
+
+        btc = ctx.get_service("bitcoin")
+        seq = ctx.get_service("sequencer")
+        reth = ctx.get_service("reth")
+
+        seqrpc = seq.create_rpc()
+        btcrpc: BitcoindClient = btc.create_rpc()
+        rethrpc = reth.create_rpc()
+
+        btc_url = btcrpc.base_url
+        btc_user = btc.props["rpc_user"]
+        btc_password = btc.props["rpc_password"]
+
+        self.logger.debug(f"BTC URL: {btc_url}")
+        self.logger.debug(f"BTC user: {btc_user}")
+        self.logger.debug(f"BTC password: {btc_password}")
+
+        web3: Web3 = reth.create_web3()
+        web3.eth.default_account = web3.address
+        el_address = web3.address
+        self.logger.debug(f"EL address: {el_address}")
+
+        # Get operators pubkey and musig2 aggregates it
+        bridge_pk = get_bridge_pubkey(seqrpc)
+        self.logger.debug(f"Bridge pubkey: {bridge_pk}")
+
+        seq_addr = seq.get_prop("address")
+        self.logger.debug(f"Sequencer Address: {seq_addr}")
+        bridge_pk = get_bridge_pubkey(seqrpc)
+
+        # Generate Plenty of BTC to address
+        btcrpc.proxy.generatetoaddress(102, address)
+
+        # Deposit to the EL address
+        # NOTE: we need 2 deposits to make sure we have funds for gas
+        self.make_drt(ctx, el_address, bridge_pk)
+        self.make_drt(ctx, el_address, bridge_pk)
+        time.sleep(0.5)
+
+        # Send funds to the bridge precompile address for a withdrawal
+        change_address_pk = extract_p2tr_pubkey(withdraw_address)
+        self.logger.debug(f"Change Address PK: {change_address_pk}")
+        l2_tx_hash = self.make_withdraw(ctx, change_address_pk, gas).hex()
+        self.logger.debug(f"Sent withdrawal transaction with hash: {l2_tx_hash}")
+
+        # Wait for the withdrawal to be processed
+        time.sleep(DEFAULT_BLOCK_TIME_SEC * 2)
+
+        l2_tx_state = web3.eth.get_transaction_receipt(l2_tx_hash)
+        self.logger.debug(f"Transaction state (web3): {l2_tx_state}")
+
+        # Make sure that the balance is zero
+        balance = int(rethrpc.eth_getBalance(el_address), 16)
+        self.logger.debug(f"Strata Balance after withdrawal: {balance}")
+
+        # lower bound is 10 BTC - 0.01 BTC (fees) - gas with a leeway
+        lower_bound = ((rollup_deposit_amount - 1_000_000) * SATS_TO_WEI) - (2 * gas * GWEI_TO_WEI)
+        self.logger.debug(f"Strata Balance lower bound: {lower_bound}")
+
+        assert balance >= lower_bound, "balance is not in the expected range"
+
+        # Mine blocks
+        btcrpc.proxy.generatetoaddress(12, UNSPENDABLE_ADDRESS)
+        time.sleep(1)
+
+        # Make sure that the balance in the BTC wallet is 10 BTC-ish
+        btc_balance = get_balance(withdraw_address, btc_url, btc_user, btc_password)
+        self.logger.debug(f"BTC balance: {btc_balance}")
+        btc_expected_balance = rollup_deposit_amount - 100_000_000
```

Comment:
Should we shift out these constants to somewhere else?

Reply:
I don't know this is just used in this withdraw case. I can make it better by using relative amounts like

```py
0.9 * rollup_deposit_amount
# or
rollup_deposit_amount // 1.1
```

Reply:
I've cleaned them up in 7a414eca5abf79b72ed921d945c06301005bca77
and also added tighter margins for lower bounds

Reply:
I've cleaned them up in 7a414eca5abf79b72ed921d945c06301005bca77
and also added tighter margins for lower bounds